### PR TITLE
docs(samples): add quickstart sample

### PR DESCRIPTION
**Pull request by stephaniewang526**
Fixes b/176624168

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/bigqueryreservation/QuickstartSample.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2021 Google Inc.
```
Comment (lesv) [inline samples/snippets/src/main/java/com/example/bigqueryreservation/QuickstartSample.java:2]: LLC

File: samples/snippets/src/test/java/com/example/bigqueryreservation/QuickstartSampleIT.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021 Google Inc.
```
Comment (lesv) [inline samples/snippets/src/test/java/com/example/bigqueryreservation/QuickstartSampleIT.java:2]: LLC

**Review by lesv — APPROVED** [review]

File: samples/snippets/src/main/java/com/example/bigqueryreservation/QuickstartSample.java (hunk 1 above)
Comment (stephaniewang526) [reply samples/snippets/src/main/java/com/example/bigqueryreservation/QuickstartSample.java]: done

File: samples/snippets/src/test/java/com/example/bigqueryreservation/QuickstartSampleIT.java (hunk 2 above)
Comment (stephaniewang526) [reply samples/snippets/src/test/java/com/example/bigqueryreservation/QuickstartSampleIT.java]: done

**Pull request merged**